src/app/application.cpp

```diff
@@ -680,6 +680,24 @@ void Application::sendNotificationEmail(const BitTorrent::Torrent *torrent)
                      content);
 }
 
+void Application::sendTestEmail() const
+{
+    const Preferences *pref = Preferences::instance();
+    if (pref->isMailNotificationEnabled())
+    {
+        // Prepare mail content
+        const QString content = tr("This is a test email.") + u'\n'
+            + tr("Thank you for using qBittorrent.") + u'\n';
+
+        // Send the notification email
+        auto *smtp = new Net::Smtp();
```

Comment:
You're not passing a `parent` to the constructor, so this is going to leak.

Reply:
Nevermind - `Smtp` is set to call `deleteLater` when its socket disconnects.

Reply:
>You're not passing a parent to the constructor, so this is going to leak.

(clarification: the code was from #20194) You are probably right. It will happen when the procedure is still running and user exited qbt. Also `Smtp` destructor seems to be missing closing the socket.
Do you mind submitting a patch?

Also note this:
https://github.com/qbittorrent/qBittorrent/blob/c06817f4ebf02ce05136916f66d5819d146b8594/src/base/net/smtp.cpp#L122

Reply:
> It will happen when the procedure is still running and user exited qbt.

What will happen?


> Also `Smtp` destructor seems to be missing closing the socket.

Socket is closed in its destructor which is called by Smtp destructor.

Reply:
But I still find it a confusing design when an explicitly created object destroys itself "without the knowledge of its creator".